birdhouse/components/weaver/post-docker-compose-up

```diff
@@ -0,0 +1,149 @@
+#!/bin/sh
+
+# Registers WPS providers into Weaver to provide their OGC-API interface
+#
+# The script first authenticates in Magpie to obtain authorization for requests, as everything could be protected.
+# Then, each provider defined in 'WEAVER_WPS_PROVIDERS' is registered iteratively.
+# All requests have a timeout to avoid hanging indefinitely.
+# Also, maximum total timeout configured by 'WEAVER_WPS_PROVIDERS_MAX_TIME' will break out if nothing responds.
+#
+# This script is executed after 'docker-compose up' of the full bird-house stack.
+# The reason why WPS providers must be registered after starting everything is because Weaver attempts to query
+# them (with GetCapabilities requests) to obtain their metadata.
+# Because some WPS containers could take longer to start, it is otherwise not guaranteed that they would be ready
+# to receive those requests before Weaver starts querying them.
+#
+# Parameters:
+#
+#   WEAVER_WPS_PROVIDERS:
+#       list of provider names (comma or space delimited), all are assumed to be available at
+#       "https://${PAVICS_FQDN_PUBLIC}${TWITCHER_PROTECTED_PATH}/<provider-name>"
+#   WEAVER_WPS_PROVIDERS_MAX_TIME:
+#       limit script execution up to maximum this number of seconds
+#
+# Following configurations are expected to be inherited from bird-house/weaver-component env.local/default.env:
+#   - MAGPIE_ADMIN_USERNAME
+#   - MAGPIE_ADMIN_PASSWORD
+#   - PAVICS_FQDN_PUBLIC
+#   - TWITCHER_PROTECTED_PATH
+#   - WEAVER_MANAGER_NAME
+#
+
+MAGPIE_URL="https://${PAVICS_FQDN_PUBLIC}/magpie"
+WEAVER_URL="https://${PAVICS_FQDN_PUBLIC}${TWITCHER_PROTECTED_PATH}/${WEAVER_MANAGER_NAME}"
+WEAVER_WPS_PROVIDERS_MAX_TIME=${WEAVER_WPS_PROVIDERS_MAX_TIME:-60}
+
+# logging
+YELLOW=${YELLOW:-$(tput setaf 3)}
+RED=${RED:-$(tput setaf 1)}
+NORMAL=${NORMAL:-$(tput sgr0)}
+PREFIX="[Weaver] "
+ERROR="${PREFIX}${RED}ERROR${NORMAL}: "
+WARNING="${PREFIX}${YELLOW}WARNING${NORMAL}: "
+
+if [ -z "${WEAVER_WPS_PROVIDERS}" ]; then
+    echo "${WARNING}Nothing specified in WEAVER_WPS_PROVIDERS to register WPS remote providers."
+    exit 0
+fi
+
+# registration of WPS providers require authenticated access, obtain login from Magpie
+start_time="$(date -u +%s)"
+while true; do
+    # login (output null + cookie-jar '-' redirects output cookie to variable)
+    cookie_jar=$( \
+        curl -k -s -m 2 -o /dev/null \
+             -X POST \
```

Comment:
Can you use long option formats in all scripts, easier to understand.  Is anyone of those options a timeout for `curl` to not be stuck?

Reply:
Yes. The `-m 2`. 
Actually, see new commit, I replaced it by `-m ${REQUEST_TIMEOUT}`.